src/checkers/inference/InferenceTreeAnnotator.java

```diff
@@ -160,6 +173,16 @@ public Void visitIdentifier(IdentifierTree node, AnnotatedTypeMirror identifierT
                     // another interface
                     variableAnnotator.visit(identifierType, node);
 
+                } else if (parentNode.getKind() == Kind.NEW_CLASS
+                        && ((NewClassTree) parentNode).getIdentifier() == node) {
+                    // This can happen at two locations in a NewClassTree:
+                    // (1) The type identifier of the NewClassTree, as `A` of `new @HERE A() {}`,
+                    // where `@HERE` is either an explicit annotation or a variable annotation.
+                    // (2) The type identifier on the anonymous class's extends/implements clause.
```

Comment:
Shouldn't this case already be covered by the visitNewClass annotator, line 298 below?

Reply:
For a new class tree, the corresponding CFG nodes follow the sequence "type identifier -> anony class declaration".
So it's essential that the type identifier gets annotated by `VariableAnnotator`, otherwise the default will be applied.

Reply:
I think we said that we'll look into this CFG issue separately. Can you 1) create an issue that describes the problem, 2) add a reference to that issue here, and then we can merge this PR.